src/Core/IThreadHandling.cs

```diff
@@ -53,6 +53,14 @@ public interface IThreadHandling
         /// </summary>
```

Comment:
I will create a seperate pr where I will rename RunOnUIThread to -> RunOnUIThreadAsync and RunOnUIThreadSync to RunOnUIThread. I didn't do it here as this pr already has a lot of changes and it would crowd it.